### PR TITLE
Support differentiable learning rates in RMSprop

### PR DESCRIPTION
**Pull request by vineethsaivs**
ok the pytorch looks good to me. go ahead with that.

> AI assistance was used to prepare this patch and the technical notes below.
>
> ## Issue
> Related to #141832 (RMSprop learning-rate differentiation).
>
> ## Summary
> RMSprop passes the learning rate as a scalar alpha/value argument, which rejects a tensor requiring gradients. Keep the rate in tensor arithmetic in differentiable mode and preserve the momentum snapshot needed by later backward passes.
>
> ## Checklist
> - [x] Passes changed-file lint (`spin fixlint`).
> - [x] Added/updated tests. Eight new two-step learning-rate gradchecks and the existing RMSprop parameter/state gradcheck pass on CPU. The eight new cases fail before the fix.
> - Documentation: no API signature change.
> - Benchmarks: not run; no performance claim.
>
> Validation runs repository test methods against the edited Python optimizers with installed PyTorch 2.10 CPU. Full rebuilt PyTorch and GPU suites were not run.
>
> ## BC-breaking?
> No.

### CODE REVIEW

**Comment by vineethsaivs** [comment]
ok the pytorch looks good to me. go ahead with that.

```
@pytorchbot label "release notes: optim"
```
